packages/next-on-pages/src/buildApplication/nextConfig.ts

```diff
@@ -0,0 +1,213 @@
+	},
+	amp: {
+		canonicalBase: '',
+	},
+	basePath: '',
+	sassOptions: {},
+	trailingSlash: false,
+	i18n: null,
+	productionBrowserSourceMaps: false,
+	optimizeFonts: true,
+	excludeDefaultMomentLocales: true,
+	serverRuntimeConfig: {},
+	publicRuntimeConfig: {},
+	reactProductionProfiling: false,
+	reactStrictMode: null,
+	httpAgentOptions: {
+		keepAlive: true,
+	},
+	outputFileTracing: true,
+	staticPageGenerationTimeout: 60,
+	swcMinify: true,
+	output: process.env['NEXT_PRIVATE_STANDALONE'] ? 'standalone' : undefined,
+	modularizeImports: undefined,
+	experimental: {
+		windowHistorySupport: false,
+		serverMinification: true,
+		serverSourceMaps: false,
+		caseSensitiveRoutes: false,
+		useDeploymentId: false,
+		deploymentId: undefined,
+		useDeploymentIdServerActions: false,
+		appDocumentPreloading: undefined,
+		clientRouterFilter: true,
+		clientRouterFilterRedirects: false,
+		fetchCacheKeyPrefix: '',
+		middlewarePrefetch: 'flexible',
+		optimisticClientCache: true,
+		manualClientBasePath: false,
+		cpus: Math.max(
+			1,
+			(Number(process.env['CIRCLE_NODE_TOTAL']) ||
+				(os.cpus() || { length: 1 }).length) - 1,
+		),
+		memoryBasedWorkersCount: false,
+		isrFlushToDisk: true,
+		workerThreads: false,
+		proxyTimeout: undefined,
+		optimizeCss: false,
+		nextScriptWorkers: false,
+		scrollRestoration: false,
+		externalDir: false,
+		disableOptimizedLoading: false,
+		gzipSize: true,
+		craCompat: false,
+		esmExternals: true,
+		isrMemoryCacheSize: 50 * 1024 * 1024,
+		incrementalCacheHandlerPath: undefined,
+		fullySpecified: false,
+		outputFileTracingRoot: process.env['NEXT_PRIVATE_OUTPUT_TRACE_ROOT'] || '',
+		swcTraceProfiling: false,
+		forceSwcTransforms: false,
+		swcPlugins: undefined,
+		largePageDataBytes: 128 * 1000,
+		disablePostcssPresetEnv: undefined,
+		amp: undefined,
+		urlImports: undefined,
+		adjustFontFallbacks: false,
+		adjustFontFallbacksWithSizeAdjust: false,
+		turbo: undefined,
+		turbotrace: undefined,
+		typedRoutes: false,
+		instrumentationHook: false,
+		bundlePagesExternals: false,
+		ppr:
+			process.env['__NEXT_TEST_MODE'] &&
+			process.env['__NEXT_EXPERIMENTAL_PPR'] === 'true'
+				? true
+				: false,
+		webpackBuildWorker: undefined,
+	},
+};
+
+/**
+ * Given a raw nextConfig object it extracts the data from it that we'd want to save as build metadata
+ * (for later runtime usage).
+ *
+ * @param nextConfig the raw config object obtained from a next.config.js file
+ * @returns the extracted config build metadata
+ */
+export function extractBuildMetadataConfig(
+	nextConfig: NextConfig,
+): NonNullable<NextOnPagesBuildMetadata['config']> {
+	const config: NonNullable<NextOnPagesBuildMetadata['config']> = {};
+
+	if (nextConfig.experimental) {
+		config.experimental = {};
+		config.experimental.allowedRevalidateHeaderKeys =
+			nextConfig.experimental.allowedRevalidateHeaderKeys;
+		config.experimental.fetchCacheKeyPrefix =
+			nextConfig.experimental.fetchCacheKeyPrefix;
```

Comment:
> Any reason not to just do:

no reason 😓 , thanks for noticing it

it definitely could be just `config.experimental = { ... }` I think something got lost here at some point 😅 

I can definitely do that but I think I went this route because I wanted to set the values only if they are actually present in the `nextConfig` and not setting fields to `undefined`.
Although having that shouldn't really present a problem and would produce simpler code so I might just go for `config.experimental = { ... }` as you suggested 🙂

Reply:
> I assume that the other experimental flag is not needed here?

Yes, here we collect things that effect the app's runtime behavior and the `incrementalCacheHandlerPath` only effects the build process